rust-multiaddr/src/errors.rs

```diff
@@ -0,0 +1,71 @@
+use std::{net, fmt, error, io, num};
+use cid;
+use byteorder;
+
+pub type Result<T> = ::std::result::Result<T, Error>;
+
+/// Error types
+#[derive(Debug)]
+pub enum Error {
+    UnknownProtocol,
+    UnknownProtocolString,
+    InvalidMultiaddr,
+    MissingAddress,
+    ParsingError(Box<error::Error + Send + Sync>),
```

Comment:
Can we make this type more specific since there's a finite set of errors it could be? Possibly `error_chain` would make this nicer. If this isn't your code then we can do this as another PR.